aaelf64-morello/aaelf64-morello.rst

```diff
@@ -0,0 +1,831 @@
+| Read-Only Data    | ``0x1BFBEULL``                  |
++-------------------+---------------------------------+
+
+When a Morello-capable assembler sees a ``.capinit`` instruction, it reserves a
+16-byte (128 bits) location (``fragment``) and  generates a
+``R_MORELLO_CAPINIT`` relocation for the linker to create a capability in the
+``fragment``. The assembler may use the ``fragment`` with the following format
+to give out size hints for the linker to use before processing the relocation.
+
++----------------+--------------+
+|  64-bit: empty | 64-bit: size |
++----------------+--------------+
+
+This size hint if not superseded by more accurate information will be
+incorporated into the ``capdesc`` ``size`` field.
+
+In case of position independent code (PIC), the assembler will generate a
+``R_MORELLO_LD128_GOT_LO12_NC`` relocation, which causes the linker to generate
+a 16-byte aligned, 16-byte sized entry in the ``.got`` that will be initialised
+by a ``capdesc`` entry in a capability descriptions table with the address of
+the ``.got`` entry as its ``location`` field. All information required to
+initialize the capability is self-contained in the ``capdesc`` entry, so the
+linker is not required to provide any size hints in the ``.got`` entry.
+
+Dynamic linking with Morello
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When dynamic linking, capability initialization is done by the dynamic linker as
+a result of processing one of the dynamic relocations listed in
+`Dynamic relocations table`_. For ``R_MORELLO_RELATIVE`` and
+``R_MORELLO_IRELATIVE`` relocations the static linker must write the following
+information to the fragment identified by ``r_offset``.
+
++------------------+-----------------+----------------------+
+|  64-bit: address | 56-bits: length |  8-bits: permissions |
++------------------+-----------------+----------------------+
+
+The 8-bit permission field of the fragment encodes the symbol permissions as
+below.
+
++-------------------+---------------------------------+
+| Permission        | Encoding                        |
++-------------------+---------------------------------+
+| Executable        | ``0x4ULL``                      |
++-------------------+---------------------------------+
+| Read-Write Data   | ``0x2ULL``                      |
++-------------------+---------------------------------+
+| Read-Only Data    | ``0x1ULL``                      |
++-------------------+---------------------------------+
+
+As in `Static linking with Morello`_, the linker creates a 16-byte aligned,
+16-byte sized entry in the ``.got`` for the ``R_MORELLO_LD128_GOT_LO12_NC``
+relocation generated by the assembler. However, a capability descriptions table
+is not generated to initialize the ``.got`` entry. Instead it is expected that
+the dynamic linker generate these itself based on the ``R_MORELLO_GLOB_DAT`` and
+``R_MORELLO_JUMP_SLOT`` relocations created by the static linker. The dynamic linker
```

Comment:
relocations `are` created

Reply:
I think "based on the ``R_MORELLO_GLOB_DAT`` and ``R_MORELLO_JUMP_SLOT`` relocations created by the static linker" is correct as it is. I've replaced ``these`` with ``the table``

Reply:
Sorry, I spotted one last issue: the dynamic linker generate`s`, or `will generate`.

Reply:
Done.